p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -125,30 +142,53 @@ where
         self.addresses.keys()
     }
 
-    /// Selects peer addresses for outbound connections
+    /// Selects peer addresses for outbound connections (except reserved)
     pub fn select_new_outbound_addresses(
         &self,
         pending_outbound: &BTreeSet<A>,
-        connected_outbound: &BTreeSet<A>,
+        connected_outbound_count: usize,
     ) -> Vec<A> {
+        let now = Instant::now();
         let count = MAX_OUTBOUND_CONNECTIONS
             .saturating_sub(pending_outbound.len())
-            .saturating_sub(connected_outbound.len());
+            .saturating_sub(connected_outbound_count);
 
         // TODO: Ignore banned addresses
```

Comment:
That sounds like a too-important bit to ignore for later and just throw in as an "unimportant TODO".

Reply:
I added the check for banned addresses.